src/server.js

```diff
@@ -0,0 +1,59 @@
+require('require-self-ref')
+const {checkWatcher, startNPMWatcher} = require('./watchNPM')
+const redis = require('redis')
+const logger = require('~/src/logging').logger(module)
+const config = require('~/src/config')
+const queue = require('windbreaker-service-util/queue')
+const Promise = require('bluebird')
+config.load()
+
+;(async () => {
+  let connection = null
+  let channel = null
+  let client = null
+  const AmqUrl = config.getAmqUrl()
+
+  const setup = async function () {
+    try {
+      connection = await queue.createConnection({
+        logger,
+        AmqUrl
+      })
+      channel = await connection.createChannel()
+      await channel.assertQueue(config.getQueueName())
+      client = redis.createClient(config.getRedisURL())
+      logger.info('setup successful')
+    } catch (error) {
+      logger.error(error)
+      logger.error('cleaning up and restarting setup')
+      client = null
+      if (channel) channel.close()
+      channel = null
+      if (connection) connection.close()
+      connection = null
+      await Promise.delay(1000).then(async function () {
+        await setup()
+      })
+    }
+  }
+
+  const watch = async function () {
+    let changes = null
+    const registryURL = config.getRegistryURL()
+    console.log(registryURL)
+    try {
+      changes = await startNPMWatcher({channel, client, registryURL})
+      logger.info('got changes: ' + JSON.stringify(changes))
+    } catch (error) {
+      if (changes)changes.destroy()
+      changes = null
+      logger.error(error)
+      await Promise.delay(1000).then(async function () {
+        await watch()
+      })
+    }
+  }
+
+  await setup()
+  watch()
+})()
```

Comment:
To make things easy to test, maybe create a class to manage the above logic. Something like `UpdatesPublisher` or `UpdatesProducer`. It could expose something like a `start` function that handles starts your 'loop'. 

For your unit tests, you can use something like [proxyquire](https://github.com/thlorenz/proxyquire) to replace the queue util's `createProducer` function with a stub (use [sinon](http://sinonjs.org)) that returns a mock producer you can use manipulate behavior. You can also mock out the `changes-stream` object. `sinon` can be used to spy on the functions you expect to be run and ensure you are producing the correct behavior. Make sure your test all of the different success and failure scenarios.

For your integration tests, just make sure you are correctly pushing data rabbit. You may still want to mock out the `changes-stream` object, but this time use a real producer and make sure that it's correctly publishing data that matches whatever schema you decide to use (please try to keep things consistent, maybe talk to @Epitrochoid about sharing schemas/models).

I recommend taking a look at our `ManagedConsumer` in the `windbreaker-service-util` repo as well as the tests we have around it to get an idea of how you can take your approach. 

Try to get some solid coverage (> 95%). Lines covered is not a great metric that a project is well tested, but it's a good indicator that you may have your bases covered. [nyc](https://github.com/istanbuljs/nyc) will do instrumentation and can provide metrics for you.